src/test/java/kitchenpos/application/MenuServiceTest.java

```diff
@@ -0,0 +1,251 @@
+package kitchenpos.application;
+
+import kitchenpos.domain.*;
+import kitchenpos.infra.FakeProfanityClient;
+import kitchenpos.infra.ProfanityClient;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class MenuServiceTest {
+
+    public static MenuRepository menuRepository = new InMemoryMenuRepository();
+
+    private MenuGroupRepository menuGroupRepository = MenuGroupServiceTest.menuGroupRepository;
+    private ProductRepository productRepository = ProductServiceTest.productRepository;
+
+    private ProfanityClient profanityClient = new FakeProfanityClient();
+
+    private MenuService menuService = new MenuService(menuRepository, menuGroupRepository, productRepository, profanityClient);
+
+    private MenuGroup menuGroup;
+    private Product product;
+    private MenuGroup toastMenuGroup;
+    private Menu savedSingleMenu;
+    private Menu hideMenu;
+
+    @BeforeEach
+    public void saveDummyMenu() {
+
+        menuGroup = MenuGroupServiceTest.saveMenuGroup(MenuGroupTest.create("1인 혼닭"));
+        product = ProductServiceTest.save(ProductTest.create("옛날통닭", 18000L));
+
+        MenuGroup setMenuGroup = MenuGroupServiceTest.saveMenuGroup(MenuGroupTest.create("1인메뉴"));
+        toastMenuGroup = MenuGroupServiceTest.saveMenuGroup(MenuGroupTest.create("인기 대표 토스트"));
+
+        Product 토스트 = ProductServiceTest.save(ProductTest.create("계란햄치즈토스트", 5000L));
+        Product 커피 = ProductServiceTest.save(ProductTest.create("아이스아메리카노", 3000L));
+
+        List<MenuProduct> setMenuProducts = new ArrayList();
+        List<MenuProduct> singleMenuProducts = new ArrayList();
+        setMenuProducts.add(MenuProductTest.create(토스트, 1));
+        setMenuProducts.add(MenuProductTest.create(커피, 1));
+
+        singleMenuProducts.add(MenuProductTest.create(토스트, 1));
```

Comment:
`setMenuProducts`와 `singleMenuProducts`를 생성할 때 완전한 상태로 생성할 수 있게 변경하면 어떤 장점이 있을까요?

```suggestion
        List<MenuProduct> setMenuProducts = Arrays.asList(MenuProductTest.create(토스트, 1), MenuProductTest.create(커피, 1));
        List<MenuProduct> singleMenuProducts = Arrays.asList(MenuProductTest.create(토스트, 1));
```

Reply:
완전한 상태로 생성하게되면 중간에 데이터가 변동되는 일이 없을거 같다는 생각을 하였습니다.
아무리 테스트코드라도 작성을 하다보면 데이터가 변동될 일이 있을텐데 
완전한 상태로 생성함으로서 이를 방지할수 있을거 같습니다
또한 좀더 테스트코드가 간결하여 가독성이 증가하는 효과가 있는듯합니다.